cupy/linalg/_product.py

```diff
@@ -10,7 +10,8 @@
 from cupy.linalg import _util
 
 _gu_func_matmul = _GUFunc(
-    _core.matmul, '(n?,k),(k,m?)->(n?,m?)', supports_batched=True)
+    _core.matmul, '(n?,k),(k,m?)->(n?,m?)', supports_batched=True,
+    supports_out=True)
```

Comment:
Yes. This is necessary to eliminate copy operations.  The general `out` support in `cupy._core._gufuncs._GUFunc` cannot know C-contiguous output is assumed at the cublas call in `cupy._core.matmul`.

Reply:
I think that there is no problem in normal usage. Do you think you need to take any measures?